cpp/src/arrow/flight/sql/odbc/odbc_impl/flight_sql_get_type_info_reader.cc

```diff
@@ -28,6 +28,7 @@ namespace arrow::flight::sql::odbc {
 
 using arrow::internal::checked_pointer_cast;
 using std::nullopt;
+using std::optional;
```

Comment:
not necessarily a blocker here but generally I think we avoid `using` types from `std`.

Reply:
https://github.com/apache/arrow/issues/47761